File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int res = 0, prev = 0;
        
        for(int i = 0; i < bank.length; i++){
            
            int curr = 0;
            
            for(int j = 0; j < bank[i].length(); j++){
                if(bank[i].charAt(j) == '1')    ++curr;
            }
            
            if(curr != 0){
                res += (prev * curr);
                prev = curr;
            }
        }
        
        return res;
    }
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int m = bank.size(), n = bank[0].size();
        int res = 0, prev = 0, count = 0;
        
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                if(bank[i][j] == '1')   ++count;
            }
         
            if(count != 0){
                res += (prev * count);
                prev = count;
                count = 0;
            }
            
        }
        
        return res;
    }
};